dotnet/src/SemanticKernel/AI/OpenAI/Services/AzureTextEmbeddings.cs

```diff
@@ -47,8 +48,15 @@ public async Task<IList<Embedding<float>>> GenerateEmbeddingsAsync(IList<string>
     {
         var deploymentName = await this.GetDeploymentNameAsync(this._modelId);
         var url = $"{this.Endpoint}/openai/deployments/{deploymentName}/embeddings?api-version={this.AzureOpenAIApiVersion}";
-        var requestBody = Json.Serialize(new AzureEmbeddingRequest { Input = data });
 
-        return await this.ExecuteEmbeddingRequestAsync(url, requestBody);
+        IEnumerable<Embedding<float>> embeddings = new List<Embedding<float>>();
+
+        foreach (string text in data)
+        {
+            var requestBody = Json.Serialize(new AzureEmbeddingRequest { Input = new List<string> { text } });
+            embeddings = embeddings.Concat(await this.ExecuteEmbeddingRequestAsync(url, requestBody));
```

Comment:
...then you can use `[List.AddRange(...)](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.addrange?view=net-7.0)` here, to prevent multiple list allocations.

And below, just `return embeddings;` (since List is an IList).